--- src/RubyGosu.cxx
+++ src/RubyGosu.cxx
@@ -3216,14 +3216,14 @@
 void SwigDirector_Window::update() {
   VALUE SWIGUNUSED result;
   
-  result = rb_funcall(swig_get_self(), rb_intern("update"), 0, Qnil);
+  result = rb_funcall(swig_get_self(), rb_intern("protected_update"), 0, Qnil);
 }
 
 
 void SwigDirector_Window::draw() {
   VALUE SWIGUNUSED result;
   
-  result = rb_funcall(swig_get_self(), rb_intern("draw"), 0, Qnil);
+  result = rb_funcall(swig_get_self(), rb_intern("protected_draw_2"), 0, Qnil);
 }
 
 
@@ -3231,7 +3231,7 @@
   bool c_result = SwigValueInit< bool >() ;
   VALUE SWIGUNUSED result;
   
-  result = rb_funcall(swig_get_self(), rb_intern("needs_redraw?"), 0, Qnil);
+  result = rb_funcall(swig_get_self(), rb_intern("protected_needs_redraw?"), 0, Qnil);
   bool swig_val;
   int swig_res = SWIG_AsVal_bool(result, &swig_val);
   if (!SWIG_IsOK(swig_res)) {
@@ -3246,7 +3246,7 @@
   bool c_result = SwigValueInit< bool >() ;
   VALUE SWIGUNUSED result;
   
-  result = rb_funcall(swig_get_self(), rb_intern("needs_cursor?"), 0, Qnil);
+  result = rb_funcall(swig_get_self(), rb_intern("protected_needs_cursor?"), 0, Qnil);
   bool swig_val;
   int swig_res = SWIG_AsVal_bool(result, &swig_val);
   if (!SWIG_IsOK(swig_res)) {
@@ -3260,7 +3260,7 @@
 void SwigDirector_Window::lose_focus() {
   VALUE SWIGUNUSED result;
   
-  result = rb_funcall(swig_get_self(), rb_intern("lose_focus"), 0, Qnil);
+  result = rb_funcall(swig_get_self(), rb_intern("protected_lose_focus"), 0, Qnil);
 }
 
 
@@ -3278,7 +3278,7 @@
   {
     obj0 = arg0 == Gosu::NO_BUTTON ? Qnil : LONG2NUM((&arg0)->id());
   }
-  result = rb_funcall(swig_get_self(), rb_intern("button_down"), 1,obj0);
+  result = rb_funcall(swig_get_self(), rb_intern("protected_button_down"), 1,obj0);
 }
 
 
@@ -3289,7 +3289,7 @@
   {
     obj0 = arg0 == Gosu::NO_BUTTON ? Qnil : LONG2NUM((&arg0)->id());
   }
-  result = rb_funcall(swig_get_self(), rb_intern("button_up"), 1,obj0);
+  result = rb_funcall(swig_get_self(), rb_intern("protected_button_up"), 1,obj0);
 }
 
 
@@ -3322,7 +3322,7 @@ void SwigDirector_Window::gamepad_connected(int index) {
   VALUE SWIGUNUSED result;
 
   obj0 = SWIG_From_int(static_cast< int >(index));
-  result = rb_funcall(swig_get_self(), rb_intern("gamepad_connected"), 1,obj0);
+  result = rb_funcall(swig_get_self(), rb_intern("protected_gamepad_connected"), 1,obj0);
 }
 
 
@@ -3331,7 +3331,7 @@ void SwigDirector_Window::gamepad_disconnected(int index) {
   VALUE SWIGUNUSED result;
 
   obj0 = SWIG_From_int(static_cast< int >(index));
-  result = rb_funcall(swig_get_self(), rb_intern("gamepad_disconnected"), 1,obj0);
+  result = rb_funcall(swig_get_self(), rb_intern("protected_gamepad_disconnected"), 1,obj0);
 }
 
 
@@ -3340,7 +3340,7 @@ void SwigDirector_Window::drop(std::string const &filename) {
   VALUE SWIGUNUSED result;
  
   obj0 = SWIG_From_std_string(static_cast< std::string >(filename));
-  result = rb_funcall(swig_get_self(), rb_intern("drop"), 1,obj0);
+  result = rb_funcall(swig_get_self(), rb_intern("protected_drop"), 1,obj0);
 }
 
 
@@ -11696,8 +11696,8 @@
   }
   
   SWIG_RubyInitializeTrackings();
-  rb_define_const(mGosu, "VERSION", SWIG_NewPointerObj(SWIG_as_voidptr(&Gosu::VERSION),SWIGTYPE_p_std__string, 0 ));
-  rb_define_const(mGosu, "LICENSES", SWIG_NewPointerObj(SWIG_as_voidptr(&Gosu::LICENSES),SWIGTYPE_p_std__string, 0 ));
+  rb_define_const(mGosu, "VERSION", SWIG_From_std_string(static_cast< std::string >(Gosu::VERSION)));
+  rb_define_const(mGosu, "LICENSES", SWIG_From_std_string(static_cast< std::string >(Gosu::LICENSES)));
   rb_define_const(mGosu, "MAJOR_VERSION", SWIG_From_int(static_cast< int >(1)));
   rb_define_const(mGosu, "MINOR_VERSION", SWIG_From_int(static_cast< int >(0)));
   rb_define_const(mGosu, "POINT_VERSION", SWIG_From_int(static_cast< int >(0)));
